src/Thoth.Json.Net/Converters.fs

```diff
@@ -0,0 +1,25 @@
+module Thoth.Json.Net.Converters
+
+open Newtonsoft.Json
+open Newtonsoft.Json.Linq
+
+type Converter (?isCamelCase : bool, ?extra: ExtraCoders) =
+    inherit JsonConverter()
+
+    override __.CanConvert(t) = true
+
+    override __.WriteJson(writer, value, _) = 
+        let t = value.GetType()
```

Comment:
This will fails if `value` is `None`. [See for the solution](https://github.com/thoth-org/Thoth.Json.Giraffe/commit/ea5202a7965ebb0188675fb3e43cd5953d09bd98)

Reply:
Hmm.
If I write ``override __.WriteJson(writer, (value : 'T), _) =``, value must be an object and can't be a value type anymore.

Reply:
@MangelMaxime , please ignore last comment. the value must be an obj, otherwise it cannnot be represented as Json. And for the very same reason the value cannot be simply ``None``. Simplest value should be ``[| None |]`.`
 

However, it still doesn't work . If code is changed to
```
  override __.WriteJson(writer, (value:'t), _) = 
        let encoder = Encode.Auto.generateEncoderCached<'t>(?isCamelCase=isCamelCase, ?extra=extra)
        (encoder value).WriteTo(writer)
        writer.Flush()
```
WriteTo fails in tests with ``System.ArgumentException: "Could not determine JSON object type for type Record9."``